### PR TITLE
Add Windows support for Arm64 HW Intrinsics

### LINKED ISSUE #18175: [Arm64] Enable HW intrinsics for Windows
It appears arm64 HW intrinsics capabilities are enabled for Linux via `PAL_GetJitCpuCapabilityFlags` and not via a cross-platform mechanism used by Windows.

Enable Windows/Linux parity by enabling (and testing) the HW intrinsics for Windows as well.

### LINKED ISSUE #20088: [Linux/arm64] Assertion failed 'featureSIMD'
With `COMPlus_FeatureSIMD=0`

https://ci.dot.net/job/dotnet_coreclr/job/master/view/arm64/job/jitstress/job/arm64_cross_checked_ubuntu16.04_jitnosimd_tst/2/consoleFull

```
[/home/robox/j/workspace/dotnet_coreclr/master/jitstress/arm64_cross_checked_ubuntu16.04_jitnosimd_tst/bin/tests/Linux.arm64.Checked/JIT/HardwareIntrinsics/Arm64/Crypto/Crypto.sh]: 
17:34:03 
17:34:03 
17:34:03 Assert failure(PID 1545 [0x00000609], Thread: 1545 [0x0609]): Assertion failed 'featureSIMD' in '<>c:<TestAes>b__6_0(struct,struct,struct):struct:this' (IL size 8)
17:34:03 
17:34:03     File: /mnt/j/workspace/dotnet_coreclr/master/arm64_cross_checked_ubuntu16.04/src/jit/simd.cpp Line: 124
17:34:03     Image: /home/robox/j/workspace/dotnet_coreclr/master/jitstress/arm64_cross_checked_ubuntu16.04_jitnosimd_tst/bin/tests/Linux.arm64.Checked/Tests/Core_Root/corerun
17:34:03 
17:34:03 /home/robox/j/workspace/dotnet_coreclr/master/jitstress/arm64_cross_checked_ubuntu16.04_jitnosimd_tst/bin/tests/Linux.arm64.Checked/JIT/HardwareIntrinsics/Arm64/Crypto/Crypto.sh: line 244:  1545 Aborted                 (core dumped) $_DebuggerFullPath \"$CORE_ROOT/corerun\" $ExePath $CLRTestExecutionArguments
17:34:03 /n/nReturn code:      1/nRaw output file:      /home/robox/j/workspace/dotnet_coreclr/master/jitstress/arm64_cross_checked_ubuntu16.04_jitnosimd_tst/bin/tests/Linux.arm64.Checked/JIT/HardwareIntrinsics/Reports/JIT.HardwareIntrinsics/Arm64/Crypto/Crypto.output.txt/nRaw output:/nBEGIN EXECUTION
17:34:03 /home/robox/j/workspace/dotnet_coreclr/master/jitstress/arm64_cross_checked_ubuntu16.04_jitnosimd_tst/bin/tests/Linux.arm64.Checked/Tests/Core_Root/corerun Crypto.exe
17:34:03 System.Runtime.Intrinsics.Arm.Arm64.Aes.IsSupported = True
17:34:03 System.Runtime.Intrinsics.Arm.Arm64.Sha1.IsSupported = True
17:34:03 System.Runtime.Intrinsics.Arm.Arm64.Sha2.IsSupported = True
17:34:03 Running tests
17:34:03 Expected: 100
17:34:03 Actual: 134
17:34:03 END EXECUTION - FAILED
17:34:03 Test Harness Exitcode is : 1
17:34:03 /nTo run the test:/n> set CORE_ROOT=/home/robox/j/workspace/dotnet_coreclr/master/jitstress/arm64_cross_checked_ubuntu16.04_jitnosimd_tst/bin/tests/Linux.arm64.Checked/Tests/Core_Root/n> /home/robox/j/workspace/dotnet_coreclr/master/jitstress/arm64_cross_checked_ubuntu16.04_jitnosimd_tst/bin/tests/Linux.arm64.Checked/JIT/HardwareIntrinsics/Arm64/Crypto/Crypto.sh/n
17:34:03 Expected: True
17:34:03 Actual:   False
```

### PR DESCRIPTION
**Pull request by CarolEidt**
Also, fix Crypto test and ensure that featureSIMD is taken into account

Fix #18175
Fix #20088

### CODE REVIEW

**Comment by CarolEidt** [comment]
There's a lot of code duplication between the x86 and arm64 implementations, and I'd like to see some unification but I'll leave that for a future PR

File: src/jit/hwintrinsicArm64.h [hunk 1]
```diff
@@ -54,6 +54,16 @@ struct HWIntrinsicInfo
     static const HWIntrinsicInfo& lookup(NamedIntrinsic id);
     static int lookupNumArgs(const GenTreeHWIntrinsic* node);
 
+    static bool isFullyImplementedIsa(InstructionSet isa);
+    static bool isScalarIsa(InstructionSet isa);
+
+    // Member lookup
+
+    static NamedIntrinsic lookupId(NamedIntrinsic id)
```
Comment (mikedn) [inline src/jit/hwintrinsicArm64.h]: This looks strange, what is this used for?

Comment (CarolEidt) [reply src/jit/hwintrinsicArm64.h]: I think for dumping but I'm not actually sure. I just copied it over from the x86 implementation with the others

**Comment by CarolEidt** [comment]
@dotnet/jit-contrib PTAL

**Pull request merged**